### PR TITLE
Add null checks on  role mapping

### LINKED ISSUE #3776: Missing null checks on roleMappings causing 500
We are not performing proper validation on those endpoints:
https://github.com/Apicurio/apicurio-registry/blob/006630fcdfbb0718cb987d276ddc09bb9a5020f4/app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java#L340

and, passing valid Json objects with empty content crashes the server with a 500 response.

### PR DESCRIPTION
**Pull request by Amoncy**
Hi @andreaTP and @EricWittmann,

This PR closes #3776 .

I have added the missing null checks on role mapping. Could you please review the changes?
If the changes are correct, I'll proceed with implementing the test cases.

Thank you

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java [hunk 1]
```diff
@@ -337,6 +337,9 @@ public RoleMapping getRoleMapping(String principalId) {
     @Authorized(style=AuthorizedStyle.None, level=AuthorizedLevel.Admin)
     @RoleBasedAccessApiOperation
     public void updateRoleMapping(String principalId, UpdateRole data) {
+        requireParameter("principalId", principalId);
+        requireParameter("roleName", data.getRole().name());
```
Comment (EricWittmann) [inline app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java:341]: But what if the `role` field is not present or is empty?  We need to check for that.

Also need a unit test I think.

**Review by EricWittmann — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java (hunk 1 above)
Comment (andreaTP) [reply app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java:341]: Correct, here the NPE is thrown by `data.getRole()` returning `null`.
And I think we need tests 👍 , I hope there is already some infrastructure to mock rbac etc.

Comment (Amoncy) [reply app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java]: Thank you for the review @EricWittmann and @andreaTP .
 I'll add null check for Role and add Unit Test as well.

**Comment by Amoncy** [comment]
Hi,

I have added the check for Role and added the corresponding Unit Tests.

Please Note: I have removed the check for "name" as I realised while going through the code, that it belongs to enum which doesn't have a possibility of a null or empty value. Hence, I haven't added any test case for the same.

Please let me know your views on this.
Thank you

File: app/src/test/java/io/apicurio/registry/rbac/AdminResourceTest.java [hunk 2]
```diff
@@ -720,6 +720,16 @@ public void testRoleMappings() throws Exception {
                 .body("error_code", equalTo(404))
                 .body("message", equalTo("No mapping for principal 'UnknownPrincipal' and role 'READ_ONLY' was found."));
 
+        //Update a mapping with null RoleType
```
Comment (andreaTP) [inline app/src/test/java/io/apicurio/registry/rbac/AdminResourceTest.java:723]: 👍

**Comment by EricWittmann** [comment]
Thanks @Amoncy - this looks good.  @andreaTP is right that we need to review basically our entire REST layer and add input checks.  I think that might be tracked already in GH as a separate issue.  If not we should add it to the 3.0 roadmap.  But it's out of scope for this PR for sure.  :)

File: app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java [hunk 3]
```diff
@@ -337,6 +337,8 @@ public RoleMapping getRoleMapping(String principalId) {
     @Authorized(style=AuthorizedStyle.None, level=AuthorizedLevel.Admin)
     @RoleBasedAccessApiOperation
     public void updateRoleMapping(String principalId, UpdateRole data) {
+        requireParameter("principalId", principalId);
+        requireParameter("role", data.getRole());
```
[the remark this answers is not in the document]
Comment (Amoncy) [reply app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java:341]: Thank you @andreaTP for the review. 
I'll look for related beginner friendly issues and add the checks there as well.

**Comment by Amoncy** [comment]
> Thanks @Amoncy - this looks good. @andreaTP is right that we need to review basically our entire REST layer and add input checks. I think that might be tracked already in GH as a separate issue. If not we should add it to the 3.0 roadmap. But it's out of scope for this PR for sure. :)

Thanks Eric for the review.

**Pull request merged**